apps/console/src/features/identity-providers/components/settings/attribute-settings.tsx

```diff
@@ -226,61 +228,84 @@ export const AttributeSettings: FunctionComponent<AttributeSelectionPropsInterfa
+            if (isEmpty(selectedClaimsWithMapping?.find(
+                (element: IdentityProviderCommonClaimMappingInterface) => element.mappedValue === roleClaimUri))
+            ) {
                 setRoleClaimUri("");
             }
 
             // Set subject URI.
-            if (isEmpty(selectedClaimsWithMapping?.find(element => element.mappedValue === subjectClaimUri))) {
+            if (isEmpty(selectedClaimsWithMapping?.find(
+                (element: IdentityProviderCommonClaimMappingInterface) => element.mappedValue === subjectClaimUri))
+            ) {
                 setSubjectClaimUri("");
             }
         }
     }, [ selectedClaimsWithMapping ]);
 
-    const handleAttributesUpdate = () => {
+    const handleAttributesUpdate = (): void => {
 
         setIsSubmitting(true);
 
-        let canSubmit = true;
+        let canSubmit: boolean = true;
         const claimConfigurations: IdentityProviderClaimsInterface = { ...initialClaims };
 
         // Prepare claim mapping for submission.
-        if (!isEmpty(selectedClaimsWithMapping?.filter(element => isEmpty(element.mappedValue)))) {
+        if (!isEmpty(selectedClaimsWithMapping?.filter(
+            (element: IdentityProviderCommonClaimMappingInterface) => isEmpty(element.mappedValue)))
+        ) {
             canSubmit = false;
         }
-        claimConfigurations["mappings"] = selectedClaimsWithMapping.map(element => {
-            return {
-                idpClaim: element.mappedValue,
-                localClaim: element.claim
-            } as IdentityProviderClaimMappingInterface;
-        });
+        claimConfigurations["mappings"] = selectedClaimsWithMapping.map(
+            (element: IdentityProviderCommonClaimMappingInterface) => {
+                return {
+                    idpClaim: element.mappedValue,
+                    localClaim: element.claim
+                } as IdentityProviderClaimMappingInterface;
+            }
+        );
 
         if (provisioningAttributesEnabled) {
             // Prepare provisioning claims for submission.
-            if (!isEmpty(selectedProvisioningClaimsWithDefaultValue?.filter(element => isEmpty(element.mappedValue)))) {
+            if (!isEmpty(selectedProvisioningClaimsWithDefaultValue?.filter(
+                (element: IdentityProviderCommonClaimMappingInterface) => isEmpty(element.mappedValue)))
+            ) {
                 canSubmit = false;
             }
-            claimConfigurations[ "provisioningClaims" ] = selectedProvisioningClaimsWithDefaultValue.map(element => {
-                return {
-                    claim: element.claim,
-                    defaultValue: element.mappedValue
-                } as IdentityProviderProvisioningClaimInterface;
-            });
+            claimConfigurations[ "provisioningClaims" ] = selectedProvisioningClaimsWithDefaultValue.map(
```

Comment:
We might need to store `provisioningClaims` in a constant.

Reply:
Since this is used only once in the file, do we really need to store this in a constant?

Reply:
Seems like these have already been defined in `IdentityProviderClaimsInterface`. Shall we use the dot notation instead of the bracket notation for better type safety? WDYT?

```suggestion
            claimConfigurations?.provisioningClaims = selectedProvisioningClaimsWithDefaultValue.map(
```

Reply:
Addressed with 3c0f397ec1bd4b4d2c2e03f3f1796a004b1384e4